code/modules/custom_ka/core.dm

```diff
@@ -0,0 +1,497 @@
+	icon = 'icons/obj/kinetic_accelerators.dmi'
+	icon_state = ""
+	item_state = "kineticgun"
+	contained_sprite = 1
+	flags =  CONDUCT
+	slot_flags = SLOT_BELT|SLOT_HOLSTER
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	w_class = 3
+	origin_tech = list(TECH_MATERIAL = 2,TECH_ENGINEERING = 2)
+
+	burst = 1
+	fire_delay = 0 	//delay after shooting before the gun can be used again
+	burst_delay = 2	//delay between shots, if firing in bursts
+	move_delay = 1
+	fire_sound = 'sound/weapons/Kenetic_accel.ogg'
+	fire_sound_text = "blast"
+	recoil = 0
+	silenced = 0
+	muzzle_flash = 3
+	accuracy = 0   //accuracy is measured in tiles. +1 accuracy means that everything is effectively one tile closer for the purpose of miss chance, -1 means the opposite. launchers are not supported, at the moment.
+	scoped_accuracy = null
+	burst_accuracy = list(0) //allows for different accuracies for each shot in a burst. Applied on top of accuracy
+	dispersion = list(0)
+	reliability = 100
+
+	var/obj/item/projectile/projectile_type = /obj/item/projectile/kinetic
+
+	pin = /obj/item/device/firing_pin
+
+	sel_mode = 1 //index of the currently selected mode
+	firemodes = list()
+
+	//wielding information
+	fire_delay_wielded = 0
+	recoil_wielded = 0
+	accuracy_wielded = 0
+	wielded = 0
+	needspin = TRUE
+
+	var/build_name = ""
+
+	//Custom stuff
+	var/obj/item/custom_ka_upgrade/cells/installed_cell
+	var/obj/item/custom_ka_upgrade/barrels/installed_barrel
+	var/obj/item/custom_ka_upgrade/upgrade_chips/installed_upgrade_chip
+
+	var/damage_increase = 0 //The amount of damage this weapon does, in total.
+	var/firedelay_increase = 0 //How long it takes for the weapon to fire, in deciseconds.
+	var/range_increase = 0
+	var/recoil_increase = 0 //The amount of recoil this weapon has, in total.
+	var/cost_increase = 0 //How much energy to take per shot, in total.
+	var/cell_increase = 0 //The total increase in battery. This actually doesn't do anything and is just a display variable. Power is handled in their own parts.
+	var/capacity_increase = 0 //How much/big this frame can hold a mod.
+	var/mod_limit_increase = 0 //Maximum size of a mod this frame can take.
+	var/aoe_increase = 0
+
+	var/current_highest_mod = 0
+
+	var/is_emagged = 0
+	var/is_emped = 0
+
+	var/can_disassemble_cell = TRUE
+
+/obj/item/weapon/gun/custom_ka/examine(var/mob/user)
+	. = ..()
+	if(installed_upgrade_chip)
+		to_chat(user,"It is equipped with \the [installed_barrel], \the [installed_cell], and \the [installed_upgrade_chip].")
+	else if(installed_barrel)
+		to_chat(user,"It is equipped with \the [installed_barrel] and \the [installed_cell]. It has space for an upgrade chip.")
+	else if(installed_cell)
+		to_chat(user,"It is equipped with \the [installed_cell]. The assembly lacks a barrel installation.")
+
+	if(installed_barrel)
+		if(custom_name)
+			to_chat(user,"[custom_name] is written crudely in pen across the side, with the offical designation \"[official_name]\" is etched neatly on the side.")
```

Comment:
Crude grammar. `While the official designation X is etched neatly...` or `With the official designation X etched neatly...`